_posts/2021-07-08-how-to-upgrade-from-opendistro-to-opensearch.markdown

```diff
@@ -0,0 +1,250 @@
+
+The first approach to upgrading from Open Distro to OpenSearch is to take a [snapshot](https://docs-beta.opensearch.org/opensearch/snapshot-restore/) of your data (indices), [create an OpenSearch cluster](https://docs-beta.opensearch.org/opensearch/install/), restore the snapshot on the new cluster, and point your clients to the new host. While this approach means running two clusters in parallel, it allows you an opportunity to validate that the OpenSearch cluster is working in a way that meets your needs prior to modifying the current Open Distro cluster. 
+
+1. Verify existing cluster is green and healthy. See the [Validation](#Validation) section below for additional guidance.
+2. Take a snapshot in Open Distro.
+
+    ```
+    PUT /_snapshot/my_backup/opendistro_backup?wait_for_completion=true
+    ```
+
+3. Spin up a new cluster using OpenSearch.
+4. Restore from the snapshot taken in step 2.
+
+    ```
+    POST /_snapshot/my_backup/opendistro_backup/_restore
+    ```
+
+5. Verify the indices are searchable and indexable in OpenSearch after restore.
+
+And that’s it! Your snapshot upgrade is complete! 
+
+### Approach #2: Restart Upgrade
+
+Another approach is restarting your whole cluster while upgrading them to OpenSearch. It has downtime and traffic from clients should be stalled until the new cluster is up. 
+
+1. Verify existing cluster is green and healthy. See the [Validation](#Validation) section below for additional guidance.
+2. Disable shard allocation: During upgrade we do not want any shard movement as the cluster will be taken down.
+
+    ```
+    PUT _cluster/settings
+    {
+      "persistent": {
+        "cluster.routing.allocation.enable": "primaries"
+      }
+    }
+    ```
+
+3. Stop indexing and perform a flush: As the cluster will be taken down, indexing/searching should be stopped and `_flush` can be used to permanently store information into the index which will prevent any data loss during upgrade.
+
+    ```
+    POST /_flush
+    ```
+
+4. Shutdown all the nodes. 
+
+    ```
+    # If you are running Elasticsearch using systemd run:
+    sudo systemctl stop elasticsearch.service
+    
+    #If you are running Elasticsearch using SysV init:
+    sudo -i service elasticsearch stop
+    
+    #If you are running Elasticsearch as daemon:
+    kill $(cat pid)
+    ```
+
+5. Upgrade all the nodes (install OpenSearch):
+    1. Extract the zip in a new directory.
+    2. Copy over the data directory from Elasticsearch to OpenSearch or set `path.data` in `config/opensearch.yml` pointing to that directory.
+    3. Set `path.logs` in `config/opensearch.yml` pointing to a directory where you want to store logs.
+
+6. Verify existing cluster is still green and healthy.
+7. Re-enable shard allocation: Start each upgraded node: If the cluster has dedicated master nodes please start them first and make sure the master is elected before data nodes are started. You can monitor the health of the cluster via
```

Comment:
":" followed by another ':". A bit awkward, we may want to drop the second for a ".".

Reply:
Thats a great catch, quip to markdown messed it up. It was supposed to be 2 different points.